lua/starfall/libs_sh/ply.lua

```diff
@@ -0,0 +1,93 @@
+return function( instance )
+
+
+local checktype = instance.CheckType
+local player_methods = instance.Types.Player.Methods
+local weapon_methods = instance.Types.Weapon.Methods
+local ply_meta, punwrap = instance.Types.Player, instance.Types.Player.Unwrap
+local builtins_library = instance.env
+
+
+local function getPly( this )
+    local ent = punwrap( this )
+    
+    if ent:IsValid() then
+        return ent
+    end
+    
+    SF.Throw( "Player is not valid.", 3 )
+end
+
+
+local function canUsePlyCore( ply )
+    return ply:IsAdmin() or ply:IsSuperAdmin()
+end
+
+
+--- Gets the slow walking speed of a player. This is when you're using +WALK.
+-- @return Slow-walk speed of the player
+function player_methods:getSlowWalkSpeed()
+    checktype( self, ply_meta )
+    local ent = getPly( self )
```

Comment:
`local ply = getPly( self )` would be better
there are a few other functions that also need this change